server/lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -2,23 +2,85 @@ defmodule RealtimeWeb.RealtimeChannel do
   use RealtimeWeb, :channel
   require Logger, warn: false
 
-  def join("realtime:" <> _topic, _payload, socket) do
-    {:ok, %{}, socket}
+  alias Phoenix.PubSub
+  alias Phoenix.Socket
+  alias Phoenix.Socket.Broadcast
+  alias RealtimeWeb.ChannelsAuthorization
+
+  intercept ["INSERT", "UPDATE", "DELETE"]
+
+  def join("realtime:" <> sub_topic = topic, %{"user_token" => token}, socket)
+      when is_binary(token) do
+    with {:ok, %{"sub" => user_id}} when is_binary(user_id) <-
+           ChannelsAuthorization.authorize(token),
+         :ok <- PubSub.subscribe(Realtime.PubSub, "subscription_manager"),
+         {:ok, bin_user_id} <- Ecto.UUID.dump(user_id),
+         %Socket{assigns: %{user_id: _}, channel_pid: channel_pid} = user_socket <-
+           assign(socket, :user_id, bin_user_id),
+         :ok <-
+           Realtime.SubscriptionManager.track_topic_subscriber(%{
+             channel_pid: channel_pid,
+             topic: sub_topic,
+             user_id: bin_user_id
+           }) do
+      {:ok, user_socket}
+    else
+      _ -> {:error, %{reason: "error occurred when joining #{topic}"}}
+    end
+  end
+
+  def join("realtime:" <> _, _, socket) do
+    {:ok, socket}
+  end
+
+  def handle_out(event, %{is_rls_enabled: is_rls_enabled, users: users} = msg, socket) do
+    user_id = socket.assigns[:user_id]
+
+    if not is_rls_enabled or (is_binary(user_id) and user_id in users) do
```

Comment:
I am worried about that part, because all connection receive list of all users. At least, need add case when `is_rls_enabled` is false (or nil?)